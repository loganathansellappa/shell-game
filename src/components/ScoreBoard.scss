.score-board {
  display: flex;
  flex-direction: column;
  align-items: center;
  align-content: space-evenly;
  .item {
    color: #cccccc;
    font-size: 4rem;
    margin: 2%;
  }
}
