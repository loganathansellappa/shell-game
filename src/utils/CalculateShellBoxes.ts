import { Children } from 'react';

const calculateShellBoxes = (children) => {
  const boundingBoxes = {};

  Children.forEach(children, (child) => {
    const domNode = child.ref.current;
    if (domNode) {
      const nodeBoundingBox = domNode.getBoundingClientRect();

      boundingBoxes[child.key] = nodeBoundingBox;
    }
  });

  return boundingBoxes;
};

export default calculateShellBoxes;
