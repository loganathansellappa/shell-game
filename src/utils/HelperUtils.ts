import { ColorCodes } from '~/@types/GenericTypes';

export const shellCount = (): number => Number(process.env.SHELL_COUNT);

export const shuffle = <T>(elements: T[]): Array<T> => {
  for (let i = elements.length - 1; i > 0; i--) {
    const j = Math.floor(Math.random() * (i + 1));
    [elements[i], elements[j]] = [elements[j], elements[i]];
  }
  return [...elements];
};

export const getColors = (count: number): ColorCodes => {
  let colorCodes = {};
  for (let i = 0; i < count; i++) {
    colorCodes[`key${i}`] = `#${Math.floor(Math.random() * 16777215).toString(
      16
    )}`;
  }
  return colorCodes;
};

export enum Level {
  Beginner = 2,
  Intermediate = 4,
  Expert = 8,
}
