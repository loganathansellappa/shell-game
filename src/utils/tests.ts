import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { BrowserRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';

export const renderWithRouter = (ui: JSX.Element, { route = '/' } = {}) => {
  window.history.pushState({}, 'Test page', route);

  return {
    user: userEvent.setup(),
    ...render(ui, { wrapper: BrowserRouter }),
  };
};
export const testStore = () => {
  const mockStore = configureStore();
  const initialState = {
    gameLog: {
      gameCount: 1,
      winners: 1,
      losers: 0,
    },
  };
  const store = mockStore(initialState);
  return store;
};
