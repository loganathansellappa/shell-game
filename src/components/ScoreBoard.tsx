import './ScoreBoard.scss';
import { useSelector } from 'react-redux';
import { RootState } from '~/store';
import React from 'react';
import { useNavigate } from 'react-router-dom';

interface ScoreBoard {
  winner: boolean;
  reset: () => void;
}

const ScoreBoard = () => {
  const navigate = useNavigate();
  const totalWinners = useSelector((state: RootState) => state.gameLog.winners);
  const totalGamesPlayed = useSelector(
    (state: RootState) => state.gameLog.gameCount
  );
  const totalLosers = useSelector((state: RootState) => state.gameLog.losers);

  return (
    <div className={'score-board'}>
      <div className="item">
        <button onClick={() => navigate('/')}>Back to Home</button>
      </div>
      <div className="item">Total Games Played {totalGamesPlayed}</div>
      <div className="item">Win Count {totalWinners}</div>
      <div className="item">Lost Count {totalLosers}</div>
    </div>
  );
};

export default ScoreBoard;
