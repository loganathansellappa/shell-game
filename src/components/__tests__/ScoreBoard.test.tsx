import { renderWithRouter, testStore } from '../..//utils/tests';
import { Provider } from 'react-redux';
import ScoreBoard from '../ScoreBoard';

describe('ScoreBoard', () => {
  let element: { container: HTMLElement };
  beforeEach(() => {
    process.env = { SHELL_COUNT: '3' };
    element = renderWithRouter(
      <Provider store={testStore()}>
        <ScoreBoard />
      </Provider>
    );
  });

  test('renders ScoreBoard Page', () => {
    expect(element.container.firstChild).toMatchSnapshot();
    expect(element.container.textContent).toContain('Total Games Played 1');
  });
});
