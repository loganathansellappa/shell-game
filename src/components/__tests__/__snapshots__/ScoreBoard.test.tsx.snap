// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScoreBoard renders ScoreBoard Page 1`] = `
<div
  class="score-board"
>
  <div
    class="item"
  >
    <button>
      Back to Home
    </button>
  </div>
  <div
    class="item"
  >
    Total Games Played 
    1
  </div>
  <div
    class="item"
  >
    Win Count 
    1
  </div>
  <div
    class="item"
  >
    Lost Count 
    0
  </div>
</div>
`;
